Malmo/src/TCPServer.cpp

```diff
@@ -109,8 +109,8 @@ namespace malmo
             if (this->closing)
             {
                 new_connection.get()->getSocket().close();
-                if (scope != nullptr) 
-                    scope->release();
+                if (this->scope != nullptr) 
+                    this->scope->release(); // Release scope which can be self.
```

Comment:
By "self" you mean "Release scope, which can be itself", rather than "Release scope, which can be _us_"?